pkg/validation/app.go

```diff
@@ -103,6 +108,59 @@ func (v *Validator) validateConfig(ctx context.Context, cr v1alpha1.App) error {
 	return nil
 }
 
+func (v *Validator) validateNamespaceConfig(ctx context.Context, cr v1alpha1.App) error {
+	annotations := key.AppNamespaceAnnotations(cr)
+	labels := key.AppNamespaceLabels(cr)
+
+	if annotations == nil && labels == nil {
+		// no-op
+		return nil
+	}
+
+	var apps []v1alpha1.App
+	{
+		lo := metav1.ListOptions{
+			FieldSelector: fmt.Sprintf("metadata.name!=%s", cr.Name),
+		}
+		appList, err := v.g8sClient.ApplicationV1alpha1().Apps(cr.Namespace).List(ctx, lo)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		apps = appList.Items
+	}
+
+	for _, app := range apps {
+		if key.AppNamespace(cr) != key.AppNamespace(app) {
+			continue
+		}
+
+		targetAnnotations := key.AppNamespaceAnnotations(app)
+		if targetAnnotations != nil && annotations != nil {
+			for k, v := range targetAnnotations {
+				originalValue, ok := annotations[k]
+				if ok && originalValue != v {
+					return microerror.Maskf(validationError, "app %#q is modifying annotation %#q in the target namespace %#q from value %#q to %#q, collide to app %#q",
+						key.AppName(cr), k, key.AppNamespace(cr), originalValue, v, app.Name)
```

Comment:
```suggestion
					return microerror.Maskf(validationError, "app %#q annotation %#q for target namespace %#q collides with value %#q for app %#q",
						key.AppName(cr), k, key.AppNamespace(cr), v, app.Name)
```

This is a tricky one. I tried to make the message shorter. Also the user knows which value they set so I think we can leave that out.

This is easier to review in the unit test message IMO.

Reply:
This is much better, Thanks!